pkg/ingester/ingester.go

```diff
@@ -324,7 +325,7 @@ func (i *Ingester) append(ctx context.Context, sample *model.Sample) error {
 
 	if err := ValidateSample(sample, &i.cfg.validationConfig); err != nil {
 		level.Error(util.WithContext(ctx, util.Logger)).Log("msg", "error validating sample", "err", err)
-		return nil
+		return err
```

Comment:
This surprised me that the error wasn't being returned...

Reply:
How does this feed through to senders?  In particular, will they retry the same data?

Reply:
Looks like this was in from day zero; it should get handled correctly, as the errors are StatusBadRequest (so no retry).

I'll merge this into our dev branch and test it out first though, just to be sure.

Reply:
This has been deployed for a day or so now, with no ill effects.